### PR TITLE
Changes to allow Nitro to be integrated into the CI system

### PR DESCRIPTION
**Pull request by dreemkiller**
These changes are required to allow veracruz to run inside a docker container for the Nitro environment. This is a partner to the PR (https://github.com/veracruz-project/veracruz-docker-image/pull/10) in the veracruz-docker-image repo. They should be merged at the same time.

### CODE REVIEW

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: c0e991ddf9e530f5f93618ae9b4957b170835470
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=4d8fd156-880c-4748-a498-0d81c716b2d4%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: f0e6cce8133d0a4c1d795e3257cf45719432c243
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=2250ab19-9e7e-46d9-a32a-8135146eabe7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by ShaleXIONG — DISMISSED** [review]

**Comment by ShaleXIONG** [comment]
This is all good from my test run. I have rebased and updated the `nitro_instruction.md`.

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 95865c3c1666082527680dbeafb3452aa78c8f87
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=7314e06b-1e79-4dbc-95f5-be7608836cc8%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by veracruz-project-owner** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: Veracruz
* Commit ID: 4583760ddf7622c2ce2a07ef127f666181603444
* Result: FAILED
* [Build Logs](https://t4r5bdzqe9.execute-api.eu-west-2.amazonaws.com/Prod/buildlogs?key=d5b4383a-b5eb-4f2d-b57e-5605fe4c7dbf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: veracruz-server-test/src/main.rs [hunk 1]
```diff
@@ -1286,8 +1298,7 @@ mod tests {
 	// the policy files. so we need to replace it with the private IP of the current instance
         #[cfg(feature = "nitro")]
         {
-            let ip_string = local_ipaddress::get()
-            .expect("Failed to get local ip address");
+            let ip_string = std::env::var("TABASCO_IP_ADDRESS").expect("TABASCO_IP_ADDRESS environment variable not set");
```
Comment (dominic-mulligan-arm) [inline veracruz-server-test/src/main.rs:1301]: I think this should be something like `PROXY_ATTESTATION_SERVER_ADDRESS`, given we've moved away from Tabasco as part of the Great Renaming.

**Comment by dominic-mulligan-arm** [comment]
What is the status of this pull request?

File: NITRO_INSTRUCTIONS.markdown [hunk 2]
```diff
@@ -1,8 +1,42 @@
 # Setting up an environment for Veracruz on AWS Nitro Enclaves
 
-Start a standard EC2 instance, with M5.large, x86_64, with a security group that allows port 22 from all IPs, ports 3010 and 9090 from the VPC IPs, with an EBS volume attached (I use 64 GB for that).
+Note that veracruz relies on an AMI available in us-east-1. We have not test other region.
```
Comment (dominic-mulligan-arm) [inline NITRO_INSTRUCTIONS.markdown:3]: sp: "We have not test other region" --> "We have not tested any other region".

File: NITRO_INSTRUCTIONS.markdown [hunk 3]
```diff
@@ -1,8 +1,42 @@
...
+
+### Internet Gateway
+
+Create an internet gateway via [this link](https://console.aws.amazon.com/vpc/). 
+Then select the new gateway and attach it to the new VPC in the action menu.
+
+### Subnet
+Create a subnet via [this link](https://console.aws.amazon.com/vpc/). 
+In the VPC ID, choose the VPC we just created. 
+In the available zone choose us-east-1f (we did not test other zone).
```
Comment (dominic-mulligan-arm) [inline NITRO_INSTRUCTIONS.markdown:18]: sp: "We did not test other zone" --> "We did not test other zones".

File: NITRO_INSTRUCTIONS.markdown [hunk 4]
```diff
@@ -1,8 +1,42 @@
...
+In the IPv4 CIDR block, we are using `192.168.32.0/19`
+Once we have the new subnet, it is convenient to auto-assign public IPv4 in the action menu. 
+
+### Security Group
+Create a new security group attach to the VP we just created.
+In the inbound rules,  allows any ssh traffic to port 22 and any TCP to port 3030 and 9090.
```
Comment (dominic-mulligan-arm) [inline NITRO_INSTRUCTIONS.markdown:24]: sp: "In the inbound rules,  allows any" --> "In the inbound rules, allow any"

File: NITRO_INSTRUCTIONS.markdown [hunk 5]
```diff
@@ -20,95 +54,52 @@ sudo yum install openssl11-devel
...
 ```
 
-Now, start a shell in the container:
-```bash
-docker exec -u dermil01 -it veracruz bash
-```
+### Start Veracruz Docker
 
-Install the AWS Nitro Enclaves CLI (source: https://docs.aws.amazon.com/enclaves/latest/user/nitro-enclave-cli-install.html) (TODO: This should be done by the docker build process):
+Start a shell on the newly launch container:
```
Comment (dominic-mulligan-arm) [inline NITRO_INSTRUCTIONS.markdown:85]: sp: "newly launch" --> "newly launched"

File: veracruz-server-test/src/main.rs [hunk 6]
```diff
@@ -142,10 +142,22 @@ mod tests {
             std::env::set_var("RUST_LOG", "info,actix_server=debug,actix_web=debug");
             let _main_loop_handle = std::thread::spawn(|| {
                 let mut sys = System::new("Veracruz Proxy Attestation Server");
+                #[cfg(feature = "nitro")]
+                let ip_addr = 
+                {
+                    let ip_string = local_ipaddress::get()
+                        .expect("Failed to get local ip address");
+                    let ip_port_string = format!("{:}:3010", ip_string);
```
Comment (dominic-mulligan-arm) [inline veracruz-server-test/src/main.rs:150]: Should this port be pulled out as a constant at the top-level?

File: veracruz-server-test/src/main.rs [hunk 7]
```diff
@@ -142,10 +142,22 @@ mod tests {
             std::env::set_var("RUST_LOG", "info,actix_server=debug,actix_web=debug");
             let _main_loop_handle = std::thread::spawn(|| {
                 let mut sys = System::new("Veracruz Proxy Attestation Server");
+                #[cfg(feature = "nitro")]
+                let ip_addr =
```
Comment (dominic-mulligan-arm) [inline veracruz-server-test/src/main.rs:146]: Should we use `cfg_if` for these mutually-exclusive configuration checks?

**Comment by dreemkiller** [comment]
This PR is superseded by #146. I will close it.

**Pull request closed without merging**